### ISSUE TITLE
Support for `langmap` in custom mapping

### ISSUE THREAD

**Issue by LamprosPitsillos** [neovim/neovim#27776]
### Problem

`langmap` works as expected for neovim motions etc but it quickly becomes _unusable_ when trying to use custom `maps` since those expect their assigned  `lhs` key , that as it seems isnt affected by the `langmap` setting.
```lua
vim.opt.langmap = {
    "ΑA", "ΒB", "ΨC", "ΔD", "ΕE", "ΦF", "ΓG", "ΗH", "ΙI", "ΞJ", "ΚK", "ΛL", "ΜM", "ΝN",
    "ΟO", "ΠP", "QQ", "ΡR", "ΣS", "ΤT", "ΘU", "ΩV", "WW", "ΧX", "ΥY", "ΖZ",
    "αa", "βb", "ψc", "δd", "εe", "φf", "γg", "ηh", "ιi", "ξj", "κk", "λl", "μm", "νn",
    "οo", "πp", "qq", "ρr", "σs", "τt", "θu", "ωv", "ςw", "χx", "υy", "ζz",
}
vim.keymap.set('n', 'Z', function() vim.notify("real lua function") end,{noremap=true})
```

### Expected behavior

I would expect that all aspects of neovim respect the langmap thus making using different languages as input seamless.

* Maybe have a way to set keymaps via keycodes (Below greek alpha and latin a have the same code on the keyboard )
![image](https://github.com/neovim/neovim/assets/61395246/c26539d5-81af-4459-9dd7-a550ffbfcf1e)
* Or extend the `langmap` to affect keymaps as well

**Comment by zeertzjq** [comment]
It's not very clear what behavior you expect. Can you be a bit more specific?

**Comment by LamprosPitsillos** [comment]
In the above lua snippet i set a lang map and a mapping.
For each compilation in the lang map , first letter gets interpreted by neovim as the second one.
This behaviour does not apply to kemaps thought.
The above keymap works as expected using English characters but not when using the type set in the langmap 

I would expect that since i specified this "translation" in the langmap it would apply to the mappings set by the user.

Tldr:
Langmap does not apply to keys in custom mappings

**Comment by zeertzjq** [comment]
From my testing this already works.

**Comment by zeertzjq** [comment]
This already works for ASCII, but not for multibyte, which is a known bug: https://github.com/vim/vim/issues/297
